mobile_verifier/src/boosting_oracles/data_sets.rs

```diff
@@ -0,0 +1,482 @@
+use std::{
+    collections::HashMap,
+    path::{Path, PathBuf},
+    pin::pin,
+    sync::Arc,
+};
+
+use chrono::{DateTime, Duration, Utc};
+use file_store::{file_sink::FileSinkClient, traits::TimestampEncode, FileStore};
+use futures_util::{Stream, StreamExt, TryFutureExt, TryStreamExt};
+use helium_proto::services::poc_mobile as proto;
+use rust_decimal::prelude::ToPrimitive;
+use rust_decimal_macros::dec;
+use sqlx::{FromRow, PgPool, QueryBuilder};
+use task_manager::ManagedTask;
+use tokio::{fs::File, io::AsyncWriteExt, sync::Mutex};
+
+use crate::{boosting_oracles::assignment::HexAssignments, coverage::SignalLevel};
+
+use super::{HexAssignment, HexBoostData};
+
+pub trait DataSet: HexAssignment + Send + Sync + 'static {
+    const TYPE: DataSetType;
+
+    fn timestamp(&self) -> Option<DateTime<Utc>>;
+
+    fn update(&mut self, path: &Path, time_to_use: DateTime<Utc>) -> anyhow::Result<()>;
+
+    fn is_ready(&self) -> bool;
+}
+
+pub struct DataSetDownloaderDaemon<T, A, B, C> {
+    pool: PgPool,
+    data_set: Arc<Mutex<T>>,
+    data_sets: HexBoostData<A, B, C>,
```

Comment:
Taking a `HexBoostData` _and_ a handle to internal data of that structure to get at `T` seems like a lot of indirection.

Reply:
the downloader needs to know it's own data set (so that it can replace it when a new one is downloaded), but it also needs all three so that it can re-assign the hexes upon a new download.